src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftPluginDescriptor.java

```diff
@@ -61,13 +62,20 @@ default ListBoxModel doFillWaitUnitItems() {
     default FormValidation doTestConnection(@QueryParameter String apiURL, @QueryParameter String authToken) {
 
 
-        if (StringUtils.isEmpty(apiURL)) {
-            if(!EnvVars.masterEnvVars.containsKey(IOpenShiftPlugin.KUBERNETES_SERVICE_HOST_ENV_KEY) &&
+        if (apiURL == null || StringUtils.isEmpty(apiURL)) {
+            if(EnvVars.masterEnvVars.containsKey(IOpenShiftPlugin.KUBERNETES_SERVICE_HOST_ENV_KEY) &&
                     !StringUtils.isEmpty(EnvVars.masterEnvVars.get(IOpenShiftPlugin.KUBERNETES_SERVICE_HOST_ENV_KEY))) {
+                apiURL = EnvVars.masterEnvVars.get(IOpenShiftPlugin.KUBERNETES_SERVICE_HOST_ENV_KEY);
+
+                if(EnvVars.masterEnvVars.containsKey("KUBERNETES_SERVICE_PORT")) {
```

Comment:
create if it does not exist, otherwise use, a IOpenShiftPlugin constant like with service host